src/js/Calculator.js

```diff
@@ -0,0 +1,82 @@
+import { STRATEGY } from './const/index.js';
+import DigitStrategy from './strategies/DigitStrategy.js';
+import ModifierStrategy from './strategies/ModifierStrategy.js';
+import OperationStrategy from './strategies/OperationStrategy.js';
+import { isEmptyArray, isNull } from './utils/common.js';
+import { beforeunload } from './utils/domApi.js';
+
+export const defaultState = {
+  x: null,
+  operator: null,
+  y: null,
+};
+
+class Calculator {
+  #$calculator;
+  #$total;
+
+  constructor({ $calculator, $total }) {
+    this.#$calculator = $calculator;
+    this.#$total = $total;
+
+    this.#init();
+  }
+
+  #init() {
+    this.#render(defaultState);
+    this.#bindEvent();
+    this.#removeEvent();
+  }
+
+  #render(state) {
+    this.#mutateState(state);
+    this.#$total.innerText = this.#convertToDisplayValue();
+  }
+
+  #mutateState(newState) {
+    this.state = newState ? { ...newState } : this.state;
+  }
+
+  #convertToDisplayValue() {
+    const values = Object.values(this.state);
+    const displayValues = values.filter((v) => !isNull(v));
+    if (isEmptyArray(displayValues)) return '0';
+    return displayValues.join('');
+  }
+
+  #bindEvent() {
+    this.#$calculator.addEventListener('click', this.#handleClick.bind(this));
+  }
+
+  #removeEvent() {
+    beforeunload(() =>
+      this.#$calculator.removeEventListener(
+        'click',
+        this.#handleClick.bind(this)
+      )
+    );
+  }
+
+  #handleClick({ target }) {
+    const strategy = this.#strategyCreator(target.dataset);
+
+    try {
+      const newState = strategy?.mutateState(target, this.state);
+      this.#render(newState);
+    } catch ({ message }) {
+      alert(message);
+    }
+  }
+
+  #strategyCreator(dataset) {
+    return this.#strategyCommand[dataset?.strategy]?.();
+  }
+
+  #strategyCommand = {
+    [STRATEGY.MODIFIER]: () => ModifierStrategy,
+    [STRATEGY.DIGIT]: () => DigitStrategy,
+    [STRATEGY.OPERATOR]: () => OperationStrategy,
+  };
```

Comment:
```suggestion
  #strategyCommand = {
    [STRATEGY.MODIFIER]: ModifierStrategy,
    [STRATEGY.DIGIT]: DigitStrategy,
    [STRATEGY.OPERATOR]: OperationStrategy,
  };
```

이렇게 바로 각 인스턴스를 참조하면 안되는 이유가 있는걸까요?

Reply:
또 이 부분은 클래스 밖에 선언해도 될 것 같아요~
꼭 클래스 멤버로 있어야 할 이유는 없어보이네요

Reply:
앗 
- 이유는 없습니다! 처음에 인스턴스를 생성을 로직을 짜야겠다 생각했었는데.. 굳이 인스턴스를.. 매번 이벤트를 호출할때마다...?? 라는 생각이 들어서 급하게 `static`으로 처리하게된 히스토리가 있었습니다! 그 과정에서 인스턴스 생성을 지연하기위해 함수를 맵핑했습니다 😹 마찬가지로 객체로 수정하면서 반영했습니다! 

- 객체로 수정하면서 밖에 선언해두었습니다! 제가 의도했던 부분은 코드 작성시점과 실행시점에 다른 `strategy`를 사용하게되서.. 다른사람이 `strategy` 가 뭔지 빠르게 파악할 수 있도록 일부로 가까운 곳에 위치를 시켰었습니다.. 그런데 말씀해주신 것 처럼 굳이 클래스나 객체 안에 둘 필요는 없을 것 같아 외부로 빼냈습니다!